packages/server/src/integrations/utils.ts

```diff
@@ -440,3 +445,59 @@ export function removeEmptyFilters(filters: SearchFilters) {
   }
   return filters
 }
+
+export async function handleXml(response: any) {
+  let data
+  const rawXml = await response.text()
+  data =
+    (await xmlParser(rawXml, {
```

Comment:
I'm a little concerned about parsing untrusted XML. I'm aware of attacks like https://cheatsheetseries.owasp.org/cheatsheets/XML_External_Entity_Prevention_Cheat_Sheet.html but I don't really see any mention of external entities in the `xml2js` library (https://github.com/Leonidas-from-XIV/node-xml2js/issues?q=is%3Aissue+external+).

The underlying library that `xml2js` is using (`sax-js`) appears to have an allowlist of entities and doesn't support external or custom ones: https://github.com/isaacs/sax-js/blob/main/lib/sax.js#L355. So we're probably safe? 🤔

Reply:
So this was already in the codebase. Adria just commented that the top level function was too messy so I moved everything out. I think it's been there for like 3 years :D